app/src/features/projects/view/components/ProjectObjectives.tsx

```diff
@@ -22,17 +33,81 @@ export interface IProjectObjectivesProps {
  */
 const ProjectObjectives: React.FC<IProjectObjectivesProps> = (props) => {
   const {
-    projectForViewData: { objectives, id }
+    projectForViewData: { objectives, id },
+    codes
   } = props;
 
-  const history = useHistory();
+  const biohubApi = useBiohubApi();
+
+  const [errorDialogProps, setErrorDialogProps] = useState<IErrorDialogProps>({
+    dialogTitle: EditObjectivesI18N.editErrorTitle,
+    dialogText: EditObjectivesI18N.editErrorText,
+    open: false,
+    onClose: () => {
+      setErrorDialogProps({ ...errorDialogProps, open: false });
+    },
+    onOk: () => {
+      setErrorDialogProps({ ...errorDialogProps, open: false });
+    }
+  });
+
+  const showErrorDialog = (textDialogProps?: Partial<IErrorDialogProps>) => {
+    setErrorDialogProps({ ...errorDialogProps, ...textDialogProps, open: true });
+  };
 
   const [openEditDialog, setOpenEditDialog] = useState(false);
 
-  const handleDialogEdit = (values: IProjectObjectivesForm) => {
-    // make put request from here using values and projectId
-    setOpenEditDialog(false);
-    history.push(`/projects/${id}/details`);
+  const [objectivesDataForUpdate, setObjectivesDataForUpdate] = useState<IGetProjectForUpdateResponseObjectives>(
```

Comment:
Yeah that's my understanding.  @NickPhura, is that the correct pattern?

Reply:
Sounds right ya.

Reply:
The Form interfaces don't necessarily match the edit interfaces.

Right now they are generally quite similar, but one thing we haven't done, which would make them even more different, is incorporate code values in the backend so that the view pages don't have to iterate over codes in the frontend, which would mean that a lot of the number fields would become strings.

But that is all stuff we can optimize later.